primitives/src/v0.rs

```diff
@@ -60,13 +62,27 @@ mod collator_app {
 /// Identity that collators use.
 pub type CollatorId = collator_app::Public;
 
+#[cfg(feature = "std")]
+impl MallocSizeOf for CollatorId {
+	fn size_of(&self, _ops: &mut MallocSizeOfOps) -> usize {
+		32
+	}
+}
+
```

Comment:
AFAIK all these types do not allocate (the suggestion also applies to them)

`MallocSizeOf::size_of` does not take into account `std::mem::size_of` the struct itself:
https://docs.rs/parity-util-mem/0.8.0/parity_util_mem/trait.MallocSizeOf.html

We could also take implement `constant_size`, so that the calculation for `Vec<T>` would be constant time:
```suggestion
#[cfg(feature = "std")]
impl MallocSizeOf for CollatorId {
	fn size_of(&self, _ops: &mut MallocSizeOfOps) -> usize {
		0
	}
	fn constant_size() -> Option<usize> {
		Some(0)
	}
}
```